src/os/bluestore/BlueStore.cc

```diff
@@ -6045,10 +6045,42 @@ void BlueStore::_txc_state_proc(TransContext *txc)
       _txc_finalize_kv(txc, txc->t);
       if (!sync_transaction) {
 	if (sync_submit_transaction) {
+	  std::lock_guard<std::mutex> l(id_lock);
```

Comment:
I think a wait would work, but I suspect it'll perform better if we just fall back to committing this particular txn in kv_sync_thread in this case.  It'll happen to any other concurrent txn that needs a nid or blobid too.  But all of them will commit *with* the max update.  If we do the wait like you suggest, then they have to wait for the max increase to commit, and *then* they will commit with the batch after that.